language/testing-infra/transactional-test-runner/src/framework.rs

```diff
@@ -715,20 +725,27 @@ fn handle_known_task<'a, Adapter: MoveTestAdapter<'a>>(
     let task_name = task.name.to_owned();
     let start_line = task.start_line;
     let stop_line = task.stop_line;
-    let result = adapter.handle_command(task);
-    let result_string = match result {
-        Ok(None) => return,
-        Ok(Some(s)) => s,
-        Err(e) => format!("Error: {}", e),
+    let (result_string, cmd_var_ctx) = match adapter.handle_command(task) {
+        Ok((result_string, cmd_var_ctx)) => {
+            if let Some(s) = result_string.as_ref() {
+                assert!(!s.is_empty());
+            }
+            (result_string, cmd_var_ctx)
+        }
+        Err(e) => (Some(format!("Error: {}", e)), None),
     };
-    assert!(!result_string.is_empty());
 
-    writeln!(
-        output,
-        "\ntask {} '{}'. lines {}-{}:\n{}",
-        task_number, task_name, start_line, stop_line, result_string
-    )
-    .unwrap();
+    if let Some(s) = result_string {
```

Comment:
We are changing the behavior here when no string is returned? Is there any justification for that?

Reply:
In the previous version, if result_string is None, we just return. But in the new version, we continue to add the template var to context, then return.